auxiliary_modules/config_access/config_parser_test.py

```diff
@@ -0,0 +1,437 @@
+from dataclasses import dataclass
```

Comment:
I have not reviewed every test in this file in detail. But I am happy to _assume_ that most of the failing tests are due to bugs in `config_parser.py`, not bugs in the tests themselves. 

Therefore I am happy that we take this approach:
* Merge this PR with failing tests in `config_parser_test.py`
* Agree on a collective approach to this problem; https://mapaction.atlassian.net/browse/DATAPIPE-4
* Then create a new branch/PR to fix any bugs in the config_parser as required.

The same approach should apply to `fallback_dict_test.py`

Reply:
Now covered in this JIRA ticket https://mapaction.atlassian.net/browse/DATAPIPE-94